Add simulation time step section to converged results (#784)
ISISNeutronMuon/MDANSE: Doc/pages/H_conv.rst

atoms which appears to be sufficient to obtain enough statistics for
good static structure factor results, so that only a short MD
simulation would be required.

Simulation Time Step
~~~~~~~~~~~~~~~~~~~~

Dynamic Incoherent Structure Factor
-----------------------------------
The DISF (also the DCSF) calculation probes the dynamics of the MD trajectory at

> different time scales. For example, smaller values of :math:`q`

Reviewer: If you don't want to keep writing 
```
:math:`q`
```

Do you know about ReST [replace](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#substitutions)? 

```
.. |q| replace:: :math:`q`
```
Means you could write this as:
```suggestion
different time scales. For example, smaller values of |q|
```

Probably not necessary now this is written, but good to know.

Author: Let's leave it for now. I might use it next time I update the docs.

correspond to larger time scales while larger values of :math:`q`
correspond to smaller time scales. To obtain accurate
DISF results we therefore need smaller time steps for the larger
values of :math:`q`.

.. _figure-timestep-inc-fqt:

.. figure:: ./Pictures/time_step_conv_disf_fqt.png
   :align: center
   :width: 11.748cm
   :height: 9.393cm

   The incoherent intermediate scattering function calculated for 120 ps
   from the same MD simulation of liquid argon but with positions

>    sampled every 120 fs and 1.2 ps shown in blue and orange respectively.

Reviewer: Would put these in the same units for obvious comparison
```suggestion
   sampled every 120 fs and 1200 fs shown in blue and orange respectively.
```
or
```suggestion
   sampled every 0.12 ps and 1.2 ps shown in blue and orange respectively.
```

.. _figure-timestep-inc-sqw:

.. figure:: ./Pictures/time_step_conv_disf_sqw.png
   :align: center
   :width: 11.748cm
   :height: 9.393cm

   The dynamic incoherent structure factor calculated from a Fourier
   transform of the above incoherent intermediate scattering function
   calculated for 120 ps from a the same MD simulation of liquid argon

>    but with positions sampled every 120 fs and 1.2 ps shown in blue and

Reviewer: Again here

Author: Changed to ps

   orange respectively.

Here we run DISF calculations with a correlation frames setting of (0, 2000, 1, 1001)
and another with a correlation frames setting of (0, 20000, 10, 101) and set :math:`q` to 42 nm\ :sup:`-1`. The
MD simulation was sampled every 120 fs so by using a ``in step of`` setting

> of 10 we are effectively sampling the trajectory every 1.2 ps. For

Reviewer: And here

each frame of the correlation function the first calculation averages
over 1000 frames while the second averages over 1900 frames. The total
time of the DISF of both calculations will be the same.

As you can see from the :numref:`figure-timestep-inc-fqt` for this larger value of :math:`q` the
intermediate scattering function decays quickly.
We can see that the second calculation (orange in :numref:`figure-timestep-inc-fqt` and :numref:`figure-timestep-inc-sqw`) time steps were too long and
did not sufficiently sample the intermediate scattering function which
had already decayed to zero by the second time step. Looking at
:numref:`figure-timestep-inc-sqw` we can see how poor the sampling of the
intermediate scattering function was. The resulting DISF (orange curve in :numref:`figure-timestep-inc-sqw`) is
severely overestimated, does not provide the higher frequency results,
and does not follow the same qualitatively behaviour as first calculation
(blue curve).

Density of States
-----------------

> The density of states (DOS) is calculated by taking a Fourier transform the

Reviewer: ```suggestion
The density of states (DOS) is calculated by taking a Fourier transform of the
```

> velocity autocorrelation function (VACF). In MDANSE velocity information from

Reviewer: ```suggestion
velocity autocorrelation function (VACF). In MDANSE, velocity information from
```

your MD data files can be saved into the MDANSE trajectory file and used
in any subsequent analysis calculation. In some cases, you may not
have saved the velocity data or you may want to recalculate the velocities
from its positions. In the DOS job you have the option to calculate
velocities from the numerical derivatives of the atomic positions. The
accuracy of these velocities will be highly dependent on the time step
of the trajectory.

.. _figure-timestep-vacf:

.. figure:: ./Pictures/time_step_conv_vacf.png
   :align: center
   :width: 11.748cm
   :height: 9.393cm

>    Velocity autocorrelation function using velocities determine by

Reviewer 2: ```suggestion
   Velocity autocorrelation function using velocities determined by
```

>    numerical derivatives of atomic positions from a the same MD

Reviewer 2: ```suggestion
   numerical derivatives of atomic positions from the same MD
```

   simulation of liquid argon but with positions sampled every 120 fs and

>    1.2 ps shown in blue and orange respectively.

Reviewer: Here

.. _figure-timestep-dos:

.. figure:: ./Pictures/time_step_conv_dos.png
   :align: center
   :width: 11.748cm
   :height: 9.393cm

   The density of states calculated from a Fourier
   transform of the above velocity autocorrelation function calculated

>    from a the same MD simulation of liquid argon but with positions

Reviewer: ```suggestion
   from the same MD simulation of liquid argon but with positions
```

>    sampled every 120 fs and 1.2 ps shown in blue and orange respectively.

Reviewer: Here

Here we run DOS calculations with a correlation frames setting of (0, 2000, 1, 1001)
and another with a correlation frames setting of (0, 20000, 10, 101). We use
an interpolation order setting of 3 so that velocities will be determined
by a numerical derivative of the atomic positions. :numref:`figure-timestep-vacf`
shows that the numerical derivative for the calculation which used the
``in step of`` setting of 10 (orange curves in :numref:`figure-timestep-vacf` and :numref:`figure-timestep-dos`),

> had caused a severely underestimated VACF value at :math:`t=0`. Additionally the longer time step means that it misses the

Reviewer: ```suggestion
had caused a severely underestimated VACF value at :math:`t=0`. Additionally, the longer time step means that it misses the
```

trough seen in the blue curve of :numref:`figure-timestep-vacf` between 0
and 2000 fs. The resulting DOS (orange curve in :numref:`figure-timestep-dos`) does not provide the higher frequency

> results, has a peak at an incorrect position, and falls off towards zero far to early.

Reviewer: ```suggestion
results, has a peak at an incorrect position, and falls off towards zero far too early.
```
